app/models/concerns/orchestration/tftp.rb

```diff
@@ -82,12 +82,20 @@ def default_pxe_render(kind)
   # Adds the host to the forward and reverse TFTP zones
   # +returns+ : Boolean true on success
   def setTFTP(kind)
-    content = generate_pxe_template(kind)
+    if kind.to_s == "PXEGrub2TargetOS"
+      content = generate_pxe_template("PXEGrub2".to_sym)
+    else
+      content = generate_pxe_template(kind)
+    end
     if content
       logger.info "Deploying TFTP #{kind} configuration for #{host.name}"
       each_unique_feasible_tftp_proxy do |proxy|
         mac_addresses_for_provisioning.each do |mac_addr|
-          proxy.set(kind, mac_addr, :pxeconfig => content)
+          targetos = nil
+          if kind.to_s == "PXEGrub2TargetOS"
```

Comment:
I think it would be good to have a helper method like
```
def isPXEGrub2TargetOS
  return true if kind.to_s == "PXEGrub2TargetOS"
  false
end
```